app/src/main/java/org/p2p/wallet/striga/user/interactor/StrigaSignupDataEnsurerInteractor.kt

```diff
@@ -0,0 +1,30 @@
+package org.p2p.wallet.striga.user.interactor
+
+import timber.log.Timber
+import org.p2p.wallet.auth.interactor.MetadataInteractor
+import org.p2p.wallet.striga.signup.steps.interactor.StrigaSignupInteractor
+
+class StrigaSignupDataEnsurerInteractor(
+    private val strigaUserInteractor: StrigaUserInteractor,
+    private val strigaSignupInteractor: StrigaSignupInteractor,
+    private val metadataInteractor: MetadataInteractor,
+) {
+
+    suspend fun ensureNeededDataLoaded() {
+        if (metadataInteractor.currentMetadata == null) {
+            Timber.i("Metadata is not fetched. Trying again...")
+            metadataInteractor.tryLoadAndSaveMetadata().throwIfFailure()
+        }
+
+        if (strigaUserInteractor.isUserCreated()) {
```

Comment:
you can remove nested if statement with 
`if (strigaUserInteractor.isUserCreated()) {
    return 
}`